src/parser.ts

```diff
@@ -163,6 +163,24 @@ function findBinaryFunction(tokens: ExprElement[], fn: string) {
   }
 }
 
+// Some minuses are initially parsed as unary functions (i.e. the string "- b" is parsed an `ExprFunction` with
+// `.fn` = "−" and an `ExprIdentifier` "b" as its single argument.)
+// If any unary minus function is preceded by another token, we need to merge the pair of tokens into a binary minus
+// function.
+function findBinaryMinusFunctions(tokens: ExprElement[]) {
+  for (let i = 1; i < tokens.length; i++) {
+    const token = tokens[i];
+    if (token instanceof ExprFunction && token.fn === '−') {
+      const a = tokens[i - 1];
+      const b = token.args[0];
+
+      const args = [removeBrackets(a), removeBrackets(b)];
+      tokens.splice(i - 1, 2, new ExprFunction('−', args));
+      i -= 1;
+    }
+  }
+}
```

Comment:
i can not understand this function, this is not return and do not modify anything except function param, just loop?

Reply:
1. looks like it should return tokens as result or throw some error as in `findBinaryFunction` function?
3. redeclaring/modifying function param makes code readability so hard.